modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -117,6 +119,54 @@ def missing_point_of_contact?
         homelessness_poc_attr.blank? && (currently_homeless_attr.present? || homelessness_risk_attr.present?)
       end
 
+      def validate_form_526_service_pay!
+        validate_form_526_military_retired_pay!
+        validate_form_526_future_military_retired_pay!
+        validate_form_526_separation_pay!
+      end
+
+      def validate_form_526_military_retired_pay!
+        receiving_attr = form_attributes.dig('servicePay', 'receivingMilitaryRetiredPay')
+        future_attr = form_attributes.dig('servicePay', 'futureMilitaryRetiredPay')
+
+        return if receiving_attr.nil? || future_attr.nil?
+        return unless receiving_attr == future_attr
+
+        # EVSS does not allow both attributes to be the same value (unless that value is nil)
+        raise ::Common::Exceptions::InvalidFieldValue.new(
+          'servicePay.militaryRetiredPay',
+          form_attributes['servicePay']['militaryRetiredPay']
+        )
+      end
+
+      def validate_form_526_future_military_retired_pay!
+        future_attr = form_attributes.dig('servicePay', 'futureMilitaryRetiredPay')
+        future_explanation_attr = form_attributes.dig('servicePay', 'futureMilitaryRetiredPayExplanation')
+        return if future_attr.nil?
+
+        if future_attr == true && future_explanation_attr.blank?
+          raise ::Common::Exceptions::UnprocessableEntity.new(
+            detail: "If 'servicePay.futureMilitaryRetiredPay' is true, then " \
+                    "'servicePay.futureMilitaryRetiredPayExplanation' is required"
+          )
+        end
+      end
+
+      def validate_form_526_separation_pay!
+        validate_form_526_separation_pay_received_date!
+      end
```

Comment:
why not call this `validate_form_526_separation_pay_received_date!` method above, on line 126? Do we need another method just to call it?